email/rfc2047.c

```diff
@@ -781,13 +781,13 @@ void rfc2047_decode_addrlist(struct AddressList *al)
   struct Address *a = NULL;
   TAILQ_FOREACH(a, al, entries)
   {
-    if (a->personal && ((strstr(a->personal, "=?")) || assumed))
+    if (a->personal && ((buf_find_string(a->personal, "=?")) || assumed))
     {
-      rfc2047_decode(&a->personal);
+      rfc2047_decode(&a->personal->data);
     }
-    else if (a->group && a->mailbox && strstr(a->mailbox, "=?"))
+    else if (a->group && a->mailbox && buf_find_string(a->mailbox, "=?"))
     {
-      rfc2047_decode(&a->mailbox);
+      rfc2047_decode(&a->mailbox->data);
```

Comment:
and this are actually dangerous, but it's subtle.

`rfc2047_encode()` and `rfc2047_decode()` replace the original string with a newly allocated one.
This means that `Buffer.data` is changed, `Buffer.dsize` is wrong and `Buffer.dptr` is now dangling!

It's a bit messy, but I think the only safe thing to do is do dupe the string, encode/decode it, then overwrite the original Buffer.